### ISSUE TITLE
R2015b vs R2018x masking

### ISSUE THREAD

**Issue by agahkarakuzu** [qMRLab/qMRLab#331]
## Expected Behavior
Fit `vfa_t1` data using GUI on R2015b as it works on R2018b. 

## Actual Behavior
It does not work with R2015b. 

## Steps to Reproduce the Problem

  1. Open GUI on R2015b 
  2. Select `vfa_t1` and download example data.
  3. `Fit Data`

An exception will thrown in R2015b, here is the origin of the problem: 

![image](https://user-images.githubusercontent.com/9632322/55599499-1ee1d080-5726-11e9-8b16-9c9756ce9f56.png)

Whereas the exact same statement works on R2018b: 

![image](https://user-images.githubusercontent.com/9632322/55599513-31f4a080-5726-11e9-8e14-415070de6275.png)

On the other hand, R2015b is OK with the following expression: 

`data.(fields{1})(~logical(data.Mask)) = 0;`

Either I am blinded to see something super obvious, or there is something fishy. 

## Specifications

  - Version: v2.0.14
  - Platform: OSX 10.14.3
  - Subsystem: R2015b

**Comment by mathieuboudreau** [comment]
I'm surprised it works in 2018 with that syntax - are we sure it does what it's supposed to that way? Sounds like the data should be looped over the last dimension (different images), and the mask applied to each set of image individually.

**Comment by agahkarakuzu** [comment]
Maybe R2018 family got smart enough to interpret the intention and `deal` with it. 

But, if this is the case, then how come `data.(fields{1})(~logical(data.Mask)) = 0;` works ¯\_(ツ)_/¯

**Comment by mathieuboudreau** [comment]
*¯\\_(ツ)_/¯

**Comment by notZaki** [comment]
Multiplication and indexing work differently.
Try:
```
data = randn(128,128,1,2);
mask = true(128,128);

size(data .* mask) % Returns: [128 128 1 2]
size(data(mask)) % Returns: [16384 1] (or 128*128 by 1)
```

**Comment by agahkarakuzu** [comment]
Yet, this still won't work on R2015b: 
```diff
data = randn(128,128,1,2);
mask = true(128,128);
size(data .* mask)

- Error using  .* 
- Matrix dimensions must agree. 
```

Yes, zeroing background using ~indexes returned by a logical mask happens through vectorization, but it still needs to deal with that non-singleton dimension. Whereas the similar logic fails with `mtimes`. I've checked Matlab's release history and could not find a relevant note.

**Comment by notZaki** [comment]
I was just trying to point out that `data.(fields{1})(~logical(data.Mask)) = 0;` is not actually applying the mask correctly.

For the original problem, maybe the following works?
```
data = randn(128,128,1,2);
mask = true(128,128);
maskedData = bsxfun(@times, data, mask);
```
Might be worth a shot, although I'm not that hopeful because the docs say that this should be automatically enabled starting from R2016b [[ref](https://www.mathworks.com/help/matlab/ref/bsxfun.html)] (check "Input Arguments" section for the note)

**Comment by agahkarakuzu** [comment]
> I was just trying to point out that data.(fields{1})(~logical(data.Mask)) = 0; is not actually applying the mask correctly. 

I see your point. This convention still masks the first (let's say) slice, but would not expand it to the second one. However, by doing so, it does not throw any dimension mismatch error at the reshaping stage. 

`bsxfun` indeed solves the problem, thanks for the `implicit expansion` pointer :)  I read [this blogpost](https://blogs.mathworks.com/loren/2016/11/10/more_thoughts_about_implicit_expansion/), saying that it was started being "internally" used by Mathworks in 2015. Given that earliest product this implemented was `R2016b`, changing this block with explicit lower level statement appears to be a better approach. 

I will check how it works on Octave as well, then fix it. Testing framework may be working around this problem. 

Thanks @notZaki 💯

**Comment by agahkarakuzu** [comment]
Resolved #333
